index.js

```diff
@@ -1,57 +1,71 @@
-
+//
+// load core modules
+//
 var fs = require('fs');
 var path = require('path');
+var EventEmitter = require('events');
+var inherits = require('util').inherits;
+
+//
+// load third-party modules
+//
+var Mocha = require('mocha');
 
-var mocha = require('mocha');
+//
+// register global variables
+//
 GLOBAL.sinon = require('sinon');
-var chai = require('chai');
-var sinonChai = require('sinon-chai');
+var chai = GLOBAL.chai = require('chai');
+
+var sinonChai = GLOBAL.sinonChai = require('sinon-chai');
 chai.use(sinonChai);
+
 GLOBAL.expect = chai.expect;
 
-var Covy = function(options) {
+//
+// Declare Covy
+//
+function Covy (options) {
+  EventEmitter.call(this);
   options = options || {};
-  this.blanketOptions = options.blanket;
+  options.mocha = options.mocha || {};
+
   this.path = options.path;
   this.ext = options.ext || '.test.js';
-  this.mocha = new mocha({
-    reporter: process.env['REPORTER'] || options.reporter || 'nyan',
-    grep: process.env['GREP'] || options.grep,
-    ui: process.env['UI'] || options.ui || 'bdd',
-    growl: true
-  });
-  if (process.env['FORCE_COV'] || this.mocha.options.reporter.match('cov')) {
-    this.blanket();
-  }
-  this.files();
+
+  options.mocha.reporter = options.mocha.reporter || 'nyan';
+  options.mocha.ui = options.mocha.ui || 'bdd';
```

Comment:
`Covy` has its own options: `reporter`, `grep`, `ui`
we dont need an `options.mocha`.